File: healthcare-backend/app/src/main/java/com/healthcare/model/Appointment.java
package com.healthcare.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
public class Appointment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne @JoinColumn(name = "patient_id")
    private Patient patient;

    @ManyToOne @JoinColumn(name = "doctor_id")
    private Doctor doctor;

    private LocalDateTime appointmentDate;
    private String status; // e.g., "BOOKED", "CANCELLED", "COMPLETED"

    // getters and setters
}

File: healthcare-frontend/src/types/Appointment.ts
export interface Appointment {
  id: number;
  appointmentDate: string;   // ISO date string
  appointmentTime: string;   // e.g. "10:30 AM"
  status: string;            // e.g. "Scheduled", "Completed", "Cancelled"
  paymentStatus: boolean;    // true if paid
  doctor: {
    id: number;
    name: string;
    specialization: string;
    user:{
    id: number;
    name: string;
    role: string;
  }
  };
  patient: {
    id: number;
    name: string;
    user:{
    id: number;
    name: string;
    role: string;
  }
  };
  
}
